cmd/ci-secret-generator/main.go

```diff
@@ -127,55 +153,104 @@ func executeCommand(command string) ([]byte, error) {
 	return out, nil
 }
 
+func replaceParameter(paramName, param, template string) string {
+	return strings.ReplaceAll(template, fmt.Sprintf("$(%s)", paramName), param)
+}
+
 func updateSecrets(bwItems []bitWardenItem, bwClient bitwarden.Client) error {
```

Comment:
For the future: If you ever end up writing a function that contains 9 loops in four nesting levels, that is a very clear sign that you are not factoring your code well

Reply:
I couldn't think of a better way of doing this, could you please let me know how you would approach it?